docs/fsharp/style-guide/formatting.md

```diff
@@ -0,0 +1,505 @@
+
+// Not OK
+spam ( ham.[ 1 ] )
+```
+
+Named arguments should also not have space surrounding the `=`:
+
+```fsharp
+// OK
+let makeStreamReader x = new System.IO.StreamReader(path=x)
+
+// Not OK
+let makeStreamReader x = new System.IO.StreamReader(path = x)
+```
+
+## Formatting blank lines
+
+* Separate top-level function and class definitions with two blank lines.
+* Method definitions inside a class are separated by a single blank line.
+* Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
+* Use blank lines in functions, sparingly, to indicate logical sections.
+
+## Formatting comments
+
+Generally prefer multiple double-slash comments over ML-style block comments.
+
+```fsharp
+// Prefer this style of comments when you want
+// to express written ideas on multiple lines.
+
+(*
+    Generally avoid these kinds of comments.
+*)
+```
+
+Inline comments should capitalize the first letter.
+
+```fsharp
+let f x = x + 1 // Increment by one.
+```
+
+## Naming conventions
+
+### Use camelCase for class-bound, expression-bound and pattern-bound values and functions
+
+It is common and accepted F# style to use camelCase for all names bound as local variables or in pattern matches and function definitions.
+
+```fsharp
+// OK
+let addIAndJ i j = i + j
+
+// Bad
+let addIAndJ I J = I+J
+
+// Bad
+let AddIAndJ i j = i + j
+```
+
+Locally-bound functions in classes should also use camelCase.
+
+```fsharp
+type MyClass() =
+
+    let doSomething () =
+
+    let firstResult = ...
+
+    let secondResult = ...
+
+    member x.Result = doSomething()
+```
+
+### Use camelCase for internal and private module-bound values and functions
+
+Use camelCase for private module-bound values, including the following:
+
+* Ad hoc functions in scripts
+
+* Values making up the internal implementation of a module or type
+
+```fsharp
+let emailMyBossTheLatestResults =
+    ...
+```
+
+### Avoid underscores in names
+
+Historically, some F# libraries have used underscores in names. However, this is no longer widely accepted, partly because it clashes with .NET naming conventions. That said, some F# programmers use underscores heavily, partly for historical reasons, and tolerance and respect is important. However, be aware that the style is often disliked by others who have a choice about whether to use it.
+
+Some exceptions includes interoperating with native components, where underscores are very common.
+
+### Use standard F# operators
+
+The following operators are defined in the F# standard library and should be used instead of defining equivalents. Using these operators is recommended as it tends to make code more readable and idiomatic. Developers with a background in OCaml or other functional programming language may be accustomed to different idioms, and this list summarizes the recommended F# operators.
+
+```fsharp
```

Comment:
content inside code blocks doesn't get translated. Perhaps a table would be better?